src/components/CowCard/CowCard.js

```diff
@@ -0,0 +1,384 @@
+                <FontAwesomeIcon
+                  {...{
+                    icon: isHeartFull(i, numberOfFullHearts)
+                      ? faFullHeart
+                      : faEmptyHeart,
+                    className: classNames('heart', {
+                      'is-full': isHeartFull(i, numberOfFullHearts),
+                    }),
+                  }}
+                />
+              </li>
+            ))}
+          </ol>
+          <Tooltip
+            {...{
+              arrow: true,
+              placement: 'top',
+              title: `Check this box to put ${cow.name} into a ${huggingMachine.name} and automatically hug them at the start of every day. Requires a Hugging Machine in your inventory.`,
+            }}
+          >
+            <FormControlLabel
+              {...{
+                control: (
+                  <Checkbox
+                    {...{
+                      color: 'primary',
+                      checked: cow.isUsingHuggingMachine,
+                      onChange: e =>
+                        handleCowAutomaticHugChange &&
+                        handleCowAutomaticHugChange(e, cow),
+                    }}
+                  />
+                ),
+                disabled: !cow.isUsingHuggingMachine && !huggingMachinesRemain,
+                label: 'Hug automatically',
+              }}
+            />
+          </Tooltip>
+          <Tooltip
+            {...{
+              arrow: true,
+              placement: 'top',
+              disableFocusListener: disableBreedingControlTooltip,
+              disableHoverListener: disableBreedingControlTooltip,
+              disableTouchListener: disableBreedingControlTooltip,
+              title: isInBreedingPen
+                ? `Uncheck this box to return ${cow.name} to the regular pen.`
+                : `Check this box to move ${
+                    cow.name
+                  } to the breeding pen to mate with a ${
+                    cow.gender === genders.MALE ? 'female' : 'male'
+                  } cow.`,
+            }}
+          >
+            <FormControlLabel
+              {...{
+                control: (
+                  <Checkbox
+                    {...{
+                      color: 'primary',
+                      checked: isInBreedingPen,
+                      onChange: e =>
+                        handleCowBreedChange && handleCowBreedChange(e, cow),
+                    }}
+                  />
+                ),
+                disabled: isInBreedingPen ? false : !canBeMovedToBreedingPen,
+                label: 'Breed',
+              }}
+            />
+          </Tooltip>
+        </>
+      )}
+    </>
+  )
+}
+
+CowCardSubheader.propTypes = {
+  cow: object.isRequired,
+  cowBreedingPen: object.isRequired,
+  cowInventory: array.isRequired,
+  handleCowAutomaticHugChange: func,
+  handleCowBreedChange: func,
+  huggingMachinesRemain: bool.isRequired,
+  isCowPurchased: bool.isRequired,
+}
```

Comment:
:arrow_up: Actual new code!

Reply:
since you're moving things around, i would recommend getting every component into its own file. it can feel like a lot of overhead when creating them, but it's much easier to sort through down the road. i would look at the top level component more like a package, which has helpers and other components inside to create that thing. in this specific case it seems like `CowCard` is the top level one, and this component could be located at `CowCard/Subheader/Subheader.js` and there could be a `CowHeader/helpers.js` for things like `getCowMapById`, or that could be in `CowHeader/helpers/getCowMapById.js`, styles can be shared or local as well as needed. there's really no right or wrong way of course, but i think there are very few downsides to the extra files.